src/wp-includes/class-wp-block-scanner.php

```diff
@@ -0,0 +1,1096 @@
+				$this->last_error = static::INCOMPLETE_INPUT;
+				return false;
+			}
+
+			$has_separator = '/' === $text[ $separator_at ];
+			if ( $has_separator ) {
+				$name_at = $separator_at + 1;
+
+				if ( $name_at >= $end ) {
+					$this->state      = static::COMPLETE;
+					$this->last_error = static::INCOMPLETE_INPUT;
+					return false;
+				}
+
+				$start_of_name = $text[ $name_at ];
+				if ( 'a' > $start_of_name || 'z' < $start_of_name ) {
+					$at = $this->find_html_comment_end( $comment_opening_at, $end );
+					continue;
+				}
+
+				$name_length = 1 + strspn( $text, 'abcdefghijklmnopqrstuvwxyz0123456789-_', $name_at + 1 );
+			} else {
+				$name_at          = $namespace_at;
+				$name_length      = $namespace_length;
+				$namespace_length = 0;
+			}
+
+			if ( $name_at + $name_length >= $end ) {
+				$this->state      = static::COMPLETE;
+				$this->last_error = static::INCOMPLETE_INPUT;
+				return false;
+			}
+
+			$after_name_whitespace_at     = $name_at + $name_length;
+			$after_name_whitespace_length = strspn( $text, " \t\f\r\n", $after_name_whitespace_at );
+			$json_at                      = $after_name_whitespace_at + $after_name_whitespace_length;
+
+			if ( $json_at >= $end ) {
+				$this->state      = static::COMPLETE;
+				$this->last_error = static::INCOMPLETE_INPUT;
+				return false;
+			}
+
+			if ( 0 === $after_name_whitespace_length ) {
+				$at = $this->find_html_comment_end( $comment_opening_at, $end );
+				continue;
+			}
+
+			$has_json    = '{' === $text[ $json_at ];
+			$json_length = 0;
+
+			/*
+			 * For the final span of the delimiter it's most efficient to find the end
+			 * of the HTML comment and work backwards. This prevents complicated parsing
+			 * inside the JSON span, which cannot contain the HTML comment terminator.
+			 *
+			 * This also matches the behavior in the official block parser, though it
+			 * allows for matching invalid JSON content.
+			 */
+			$comment_closing_at = strpos( $text, '-->', $json_at );
+			if ( false === $comment_closing_at ) {
+				$this->state      = static::COMPLETE;
+				$this->last_error = static::INCOMPLETE_INPUT;
+				return false;
+			}
+
+			/*
+			 * It looks like this logic leaves an error in here, when the position
+			 * overlaps the JSON or block name. However, for neither of those is it
+			 * possible to parse a valid block if that last overlapping character
+			 * is the void flag. This, therefore, will be valid regardless of how
+			 * the rest of the comment delimiter is written.
+			 */
+			if ( '/' === $text[ $comment_closing_at - 1 ] ) {
+				$has_void_flag    = true;
+				$void_flag_length = 1;
+			} else {
+				$has_void_flag    = false;
+				$void_flag_length = 0;
+			}
+
+			/*
+			 * If there's no JSON, then the span of text after the name
+			 * until the comment closing must be completely whitespace.
+			 */
+			if ( ! $has_json ) {
+				$max_whitespace_length = $comment_closing_at - $json_at - $void_flag_length;
```

Comment:
The value of the variable is not used anywhere.

Reply:
gone!